CorsixTH/Lua/hospital.lua

```diff
@@ -936,12 +982,32 @@ function Hospital:boilerBreakdown()
   end
 end
 
--- When the boiler has been repaired this function is called.
-function Hospital:boilerFixed()
-  self.radiator_heat = self.curr_setting
-  self.heating_broke = false
-  if self:isPlayerHospital() then
-    self.world.ui.adviser:say(_A.boiler_issue.resolved)
+--! Boiler broke down and work is done to get it fixed.
+function Hospital:_fixBoiler()
+  local heat_vars = self.heating
+
+  if not heat_vars.heating_broke then return end -- Not broken, done!
+
+  -- Repair the boiler or radiators, more handyman speeds up repair, see also github #490
+  local num_radiators = self:countRadiators()
+  local num_handyman = self:countStaffOfCategory("Handyman")
+  if num_radiators < 5 * num_handyman then
+    heat_vars.boiler_repair_count = heat_vars.boiler_repair_count - 3
+  elseif num_radiators < 8 * num_handyman then
+    heat_vars.boiler_repair_count = heat_vars.boiler_repair_count - 2
+  elseif num_handyman > 0 then
+    heat_vars.boiler_repair_count = heat_vars.boiler_repair_count - 1
+  end
+  -- No repair without handyman!
```

Comment:
I have a concern that we get stuck in a state of permanent breakage here. Did this happen in the original?
I feel we should add an adviser message if this case happens (though very unlikely).

Reply:
resolved, boiler can now repair itself